examples/matrix_multiplication.masm

```diff
@@ -0,0 +1,727 @@
+# is_not_zero_or_negative_one(el: Feld) -> bool
+#
+# True if the value on the top of the stack is different than 0 and -1.
+#
+# This test is approximately 5 cycles, so 1 cycle faster than a lt operation.
+# It only works for field elements, and not u32.
+proc.is_not_zero_or_negative_one.0
+    # input: [el, ...]
+    # output: [b, ...]
+
+    # [b1, el, ...]
+    dup neq.0
+
+    # [b2, b1, ...]
+    swap neq.18446744069414584320
+
+    and
+end
+
+# count_to_word(length: Feld) -> Feld
+#
+# Given the top of the stack is a length, determines how many words are
+# required to store `length` items consecutively.
+proc.count_to_word.0
+    # input: [length, ...]
+    # output: [word_count, ...]
+    u32checked_divmod.4 eq.0 not add
+end
+
+# multiply_add_word(a: Word, b: Word) -> Feld
+#
+# Given two words on the top of the stack, first multiply elements at the same
+# position, then add the results together.
+proc.multiply_add_word.0
+    # input: [A, B, ...]
+    # output: [c, ...]
+
+    # [a1, a2, a3, a4, b1, b2, b3, b4, ...]
```

Comment:
nit: for clarity and consistency with the vm, stdlib, etc, I would write this as `[a3, a2, a1, a0, b3, b2, b1, b0, ...]`. In stack order, the first element of the word is always deepest in the stack, and we also index the elements from zero.